components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/nats/NatsProcessor.java

```diff
@@ -0,0 +1,81 @@
+package org.wso2.carbon.inbound.endpoint.protocol.nats;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.SynapseException;
+import org.apache.synapse.inbound.InboundProcessorParams;
+import org.apache.synapse.inbound.InboundTaskProcessor;
+import org.apache.synapse.task.TaskStartupObserver;
+import org.wso2.carbon.inbound.endpoint.common.InboundRequestProcessorImpl;
+import org.wso2.carbon.inbound.endpoint.common.InboundTask;
+import org.wso2.carbon.inbound.endpoint.protocol.PollingConstants;
+
+import java.util.Properties;
+
+public class NatsProcessor extends InboundRequestProcessorImpl implements TaskStartupObserver, InboundTaskProcessor {
+
+    private static final Log log = LogFactory.getLog(NatsProcessor.class.getName());
+
+    private static final String ENDPOINT_POSTFIX = "NATS" + COMMON_ENDPOINT_POSTFIX;
+
+    private NatsPollingConsumer pollingConsumer;
+    private Properties natsProperties;
+    private boolean sequential;
+    private String injectingSeq;
+    private String onErrorSeq;
+
+    public NatsProcessor(InboundProcessorParams params) {
+        this.name = params.getName();
+        this.natsProperties = params.getProperties();
+
+        String inboundEndpointInterval = natsProperties.getProperty(PollingConstants.INBOUND_ENDPOINT_INTERVAL);
+        if (inboundEndpointInterval != null) {
```

Comment:
Can this be null if the parameter does not exist in the configuration? If so, do we not need to print a warning and set the interval to a default?